assets/dev/js/editor/components/template-library/manager.js

```diff
@@ -251,6 +253,77 @@ const TemplateLibraryManager = function() {
 		} );
 	};
 
+	this.createFolder = function( folderData, options ) {
```

Comment:
It would probably be a separate task, but I wonder if we can improve the UX when the confirm button is clicked. It feels like the screen just freezes for a few seconds and then reloads. Maybe the add new folder icon could be replaced with a spinner while the request is happening (similar to the delete/rename actions).